### PR TITLE
Annotate fallthrough with TS_FALLTHROUGH

### PR DESCRIPTION
**Pull request by kev009**
Submitted by otis@freebsd.org

### CODE REVIEW

File: tsl/src/reorder.c [hunk 1]
```diff
@@ -794,6 +794,7 @@ copy_heap_data(Oid OIDNewHeap, Oid OIDOldHeap, Oid OIDOldIndex, bool verbose,
 			case HEAPTUPLE_RECENTLY_DEAD:
 				tups_recently_dead += 1;
 				/* fall through */
```
Comment (pmwkaa) [inline tsl/src/reorder.c:796]: renaming this comment to "fallthrough" could also help

Comment (kev009) [reply tsl/src/reorder.c:796]: it seems like a peculiarity of clang 10, look at this (unfortunate) thread https://bugs.llvm.org/show_bug.cgi?id=43465.  in newer version and gcc the existing comment is sufficient.

**Comment by kev009** [comment]
I went ahead and disabled -Werror for FreeBSD ports builds since such a thing is inherently fragile and not recommended for end meta build/packaging systems.  I'm indifferent to this PR now as there are some other warnings that trigger on the older clangs.

File: tsl/src/reorder.c [hunk 2]
```diff
@@ -794,6 +794,7 @@ copy_heap_data(Oid OIDNewHeap, Oid OIDOldHeap, Oid OIDOldIndex, bool verbose,
...
+				__attribute__((fallthrough));
```
Comment (nikkhils) [inline tsl/src/reorder.c:797]: Better to use the `TS_FALLTHROUGH` macro instead of this.

Comment (kev009) [reply tsl/src/reorder.c]: @nikkhils thanks that looks good, I force pushed an update to the PR

**Review by nikkhils — COMMENTED** [review]
This would require to include the "annotations.h" header, right?

**Review by k-rus — APPROVED** [review]

**Review by nikkhils — APPROVED** [review]
LGTM

**Comment by nikkhils** [comment]
@kev009 patch looks good. Please update branch and merge. Or let me know if you'd like me to merge on your behalf.

**Comment by kev009** [comment]
@nikkhils not sure what you mean I don't have write access

**Comment by svenklemm** [comment]
@kev009 Thank you for your contribution.

**Pull request merged**